smolyay/benchmark/benchmark.py

```diff
@@ -28,19 +24,19 @@ def __call__(self,x):
             If the input is outside the function domain.
         """
         x = numpy.array(x, copy=False,ndmin=2)
-        if self.dimension > 1:
-            oob = any(numpy.any(xi < bound[0]) or numpy.any(xi > bound[1]) 
-                    for xi,bound in zip(x.transpose(),self.domain))
-        else:
-            oob = (numpy.any(x < self.domain[0][0]) or 
-                    numpy.any(x > self.domain[0][1]))
+        if self.dimension == 1 and x.shape[-1] > 1:
+            x = numpy.reshape(x,list(x.shape)+[1])
```

Comment:
Doesn't this lead to a higher dimensional array than what we want? For example, if `x` is a `(5,)` list, then it will be cast to `(1,5)` by the array command, then you are adding a 1 so its shape will be `(1,5,1)`. But, we would want the shape to be `(5,1)`.

I would take the transpose if it is a 2d array meeting all the requirements:
```suggestion
        if self.dimension == 1 and x.ndim == 2 and x.shape[0] == 1 and x.shape[1] > 1:
            x = x.T
```
Then, I would add an error check on the shape of `x` before you allow the code to proceed. Specifically, that the last element of `shape` matches dimension:
```
if x.shape[-1] != self.dimension:
    raise TypeError(...)
```

Reply:
It doesn't really matter if there's an extra dimension on the end because numpy.squeeze later in the code reduces any extra dimensions. Tacking on a dimension of 1 to the end lets the code be more flexible if you say have a BenchmarkFunction object that's 1D and you want to be able to pass a n-dimensional grid.

Reply:
You shouldn't rely on something happening later to fix the input you've gotten. I didn't realize this was also a potential use case (1d function takes a multidimensional array that needs an axis added). I think the code you want then is:
```
x = numpy.array(x, copy=False, ndmin=2)
if self.dimension == 1:
    if x.ndim > 2:
        # add a new axis for the last 1d
        x = x[..., numpy.newaxis]
    elif x.shape[0] == 1 and x.shape[1]  > 1:
        # the cast to 2d puts these in wrong order, so transpose
        x = x.T
```

FYI: I looked in the documentation and transpose does not actually invoke a memory copy, which is nice! However, it is cleaner to use it sparingly.

Reply:
This doesn't account for my test case where I pass an array of size (2,3) into a 1D function, so I modified the if statements. The 1D functions really shouldn't activate the wrong number of dimensions error.